### ISSUE TITLE
Scaffold Sql Server Database fail on index with include columns

### ISSUE THREAD

**Issue by strangegt** [dotnet/efcore#17083]
Describe what is not working as expected.

When scaffold a database and  Index have include fields, they are scaffold as key columns.
The scaffold must use .ForSqlServerInclude("IIncludeColumn",...) for the include columns
                
### Steps to reproduce

This is the original index in the database

```
CREATE NONCLUSTERED INDEX [_dta_index_Clientes_54_1472724299__K2_K21_K1_K6_3_4_5_7_8_9_10_11_12_13_14_15_16_17_18_19_20_22_23_24_25_26_27_28_29_30_31_32_] ON [dbo].[Clientes]
(
	[IDTaller] ASC,
	[Referencia] ASC,
	[IDCliente] ASC,
	[Nombre] ASC
)
INCLUDE ( 	[IDComercial],
	[IDTipoCliente],
	[RazonSocial],
	[Apellidos],
	[NIFCIF],
	[Direccion],
	[CP],
	[Poblacion],
	[Provincia],
	[Pais],
	[Contacto],
	[Telefono],
	[Movil],
	[Fax],
	[Email],
	[Web],
	[IDIdioma],
	[FechaCarnet],
	[FechaNacimiento],
	[SeguroNombre],
	[SeguroApellidos],
	[SeguroNIFCIF],
	[BloqueaEnvios],
	[BajaEnvios],
	[MotivoBajaEnvios],
	[TelefonoOk],
	[MovilOk],
	[EmailOk],
	[SmsNoEmailOk],
	[EmailNoSMSOk],
	[LlamadaOk]) WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, SORT_IN_TEMPDB = OFF, DROP_EXISTING = OFF, ONLINE = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]

```

this is the scafold, it join all the columns as part of the key
```
            entity.HasIndex(e => new { e.IDComercial, e.IDTipoCliente, e.RazonSocial, e.Apellidos, e.NIFCIF, e.Direccion, e.CP, e.Poblacion, e.Provincia, e.Pais, e.Contacto, e.Telefono, e.Movil, e.Fax, e.Email, e.Web, e.IDIdioma, e.FechaCarnet, e.FechaNacimiento, e.SeguroNombre, e.SeguroApellidos, e.SeguroNIFCIF, e.BloqueaEnvios, e.BajaEnvios, e.MotivoBajaEnvios, e.TelefonoOk, e.MovilOk, e.EmailOk, e.SmsNoEmailOk, e.EmailNoSMSOk, e.LlamadaOk, e.IDTaller, e.Referencia, e.IDCliente, e.Nombre })
                    .HasName("_dta_index_Clientes_54_1472724299__K2_K21_K1_K6_3_4_5_7_8_9_10_11_12_13_14_15_16_17_18_19_20_22_23_24_25_26_27_28_29_30_31_32_");

```

### Further technical details
EF Core version: 2.2.0

Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Operating system: W10
IDE: VS 2019 16.2.1

**Comment by yyjdelete** [comment]
+1 for this, and it also makes `UNIQUE INDEX(A)` with `INCLUDE(B)` be treated as `UNIQUE INDEX(A,B)` and not work as expected a, and broken all foreign keys on it.

`Could not scaffold the foreign key 'BBB(AAA)'. A key for 'AAA' was not found in the principal entity type 'CCC'.`

**Comment by ajcvickers** [comment]
Re-opening; see #23089.

**Comment by bricelam** [comment]
This issue is still fixed in 5.0–the included columns were erroneously scaffolded as part of the indexed columns.
